File: src/ProjectR.Scripting/content/scripts/spells/SecretTechnique64.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Secret Technique #64"
end

function GetDescription()
return "Strike out at all enemies using shadow energy.\nUseful against low DEF enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .64
end

function GetDelay()
  return .1
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  local damage = (5.25 * aAD * (aDRK/100) - 1.75 * dDEF) * (100/dDRK)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/SecretTechnique64.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class SecretTechnique64 : SpellScriptBase
    {
        public override string Name { get { return "Secret Technique #64"; } }
        public override string Description { get { return "Strike out at all enemies using shadow energy.\nUseful against low DEF enemies."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 64; } }
        public override double Delay { get { return .1; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (5.25 * aAD * (aDRK / 100) - 1.75 * dDEF) * (100 / dDRK);
            DealDamage(damage);
        }
    }
}